src/wwwroot/app/views/candidate/candidate.controller.js

```diff
@@ -1,18 +1,27 @@
-export default function CandidateController($scope, CandidateService) {
+export default function CandidateController($scope, CandidateService, LoggerService) {
    'ngInject';
 
    var urlId = 'Candidates/Put';
    var vm = $scope;
    vm.candidate = {};
-
-   function addCandidate(firstName, lastName, gender) {
-      vm.candidate.firstName = firstName;
-      vm.candidate.lastName = lastName;
-      vm.candidate.gender = gender;
-      var candidateJson = JSON.stringify(vm.candidate);
-      CandidateService.addCandidate(urlId, candidateJson);
+   function _onError(message) {
+      if (message.status === -1) {
+         LoggerService.error(new Date(), 'You cannot get a response from the server');
+      }
    }
 
-
+   function addCandidate(firstName, middleName, lastName, industry, source,
+                          positionDesired, salaryDesired) {
+      vm.candidate = {};
+      vm.candidate.personalInfo = {};
+      vm.candidate.workInfo = {};
+      vm.candidate.personalInfo.firstName = firstName;
+      vm.candidate.personalInfo.middleName = middleName;
+      vm.candidate.personalInfo.lastName = lastName;
+      vm.candidate.industry = industry;
+      vm.candidate.workInfo.positionDesired = positionDesired;
+      vm.candidate.workInfo.salaryDesired = salaryDesired;
+      CandidateService.saveCandidate(vm.candidate).catch(_onError);
+   }
    vm.addCandidate = addCandidate;
```

Comment:
First of all, you need rename function to `save` or `saveCandidate`.
Also read https://github.com/johnpapa/angular-styleguide/tree/master/a1#style-y033